resources/environments/schema_gcp_credential.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 Cloudera. All Rights Reserved.
+//
+// This file is licensed under the Apache License Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied. Refer to the License for the specific
+// permissions and limitations governing your use of the file.
+
+package environments
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+)
+
+func (r *gcpCredentialResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "The GCP credential is used for authorization to provision resources such as compute instances within your cloud provider account.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"credential_name": schema.StringAttribute{
+				MarkdownDescription: "The name of the CDP credential.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"credential_key_file": schema.StringAttribute{
```

Comment:
The CLI recommends the use of local files, but I don't think we should do that for GCP in TF. We have the same problem between Azure and GCP, and we allow passing sensitive data for Azure creds. Plus, in case the credential is created by a GCP TF resource where the user will just just the output of the credential to pass it to the CDP credential object, we should be able to accept passing the creds as plain text.

Reply:
@balazsgaspar which one would you choose:
* GCP json in plain format in the TF file
* Reference the GCP json file from the TF file

Reply:
I agree - avoid passing anything like that in a local file.
The primary use of the provider in the first few months will be via the Terraform Module for AWS|Azure|GCP prerequisites, where we pre-create the GCP credential JSON using GCP's Terraform provider.

Therefore, it is even easier to pass the JSON as a variable.

So yes, definitely we should go with "_Reference the GCP json file from the TF file_"